Tests/DatadogTests/Datadog/CrashReporting/CrashContext/CrashContextProviderTests.swift

```diff
@@ -14,355 +14,128 @@ import CoreTelephony
-            rumViewEventProvider: rumViewEventProvider,
-            rumSessionStateProvider: .mockAny()
-        )
-        crashContextProvider.onCrashContextChange = { newContext in
-            updatedContext = newContext
+        crashContextProvider.onCrashContextChange = {
+            XCTAssertEqual($0.lastRUMViewEvent, viewEvent)
             expectation.fulfill()
         }
-        rumWithCrashContextIntegration.update(lastRUMViewEvent: randomRUMViewEvent)
-
-        // Then
-        waitForExpectations(timeout: 1, handler: nil)
-        XCTAssertEqual(initialContext.lastRUMViewEvent, initialRUMViewEvent)
-        XCTAssertEqual(updatedContext?.lastRUMViewEvent, randomRUMViewEvent)
-    }
-
-    // MARK: - RUM Session State Integration
-
-    func testWhenRUMWithCrashContextIntegrationIsUpdatedWithRUMSessionState_thenCrashContextProviderNotifiesNewContext() {
-        let expectation = self.expectation(description: "Notify new crash context")
-        let initialRUMSessionState: RUMSessionState = .mockRandom()
-        let randomRUMSessionState: RUMSessionState = .mockRandom()
-
-        let rumSessionStateProvider = ValuePublisher<RUMSessionState?>(initialValue: initialRUMSessionState)
-        let crashContextProvider = CrashContextProvider(
-            consentProvider: .mockAny(),
-            userInfoProvider: .mockAny(),
-            networkConnectionInfoProvider: NetworkConnectionInfoProviderMock.mockAny(),
-            carrierInfoProvider: CarrierInfoProviderMock.mockAny(),
-            rumViewEventProvider: .mockRandom(),
-            rumSessionStateProvider: rumSessionStateProvider,
-            appStateListener: AppStateListenerMock.mockAny()
-        )
-
-        let initialContext = crashContextProvider.currentCrashContext
-        var updatedContext: CrashContext?
 
-        // When
-        let rumWithCrashContextIntegration = RUMWithCrashContextIntegration(
-            rumViewEventProvider: .mockRandom(),
-            rumSessionStateProvider: rumSessionStateProvider
-        )
-        crashContextProvider.onCrashContextChange = { newContext in
-            updatedContext = newContext
-            expectation.fulfill()
-        }
-        rumWithCrashContextIntegration.update(lastRUMSessionState: randomRUMSessionState)
+        core.send(message: .custom(key: "rum", baggage: ["view-event": viewEvent]))
 
         // Then
-        waitForExpectations(timeout: 1, handler: nil)
-        XCTAssertEqual(initialContext.lastRUMSessionState, initialRUMSessionState)
-        XCTAssertEqual(updatedContext?.lastRUMSessionState, randomRUMSessionState)
+        waitForExpectations(timeout: 0.5, handler: nil)
     }
 
-    // MARK: - `UserInfo` Integration
-
-    func testWhenUserInfoValueChangesInUserInfoProvider_thenCrashContextProviderNotifiesNewContext() {
+    func testWhenRUMWithCrashContextIntegrationViewReset_thenCrashContextProviderNotifiesNewContext() {
```

Comment:
Sounds very odd 🤔💭. How about simply: _**when** RUM view is resent through message bus, **then** it notifies new crash context_?